front/src/applications/stdcm/hooks/useStdcm.ts

```diff
@@ -87,74 +93,260 @@ const useStdcm = ({
+        comfort: payload.body.comfort,
+        constraint_distribution: 'MARECO',
+        path: payload.body.steps.map((step) => ({ ...step.location, id: nextId() })),
+        rolling_stock_name: stdcmRollingStock!.name,
+        start_time: departureTime!,
+        train_name: 'stdcm',
+      };
+      dispatch(updateSelectedTrainId(STDCM_TRAIN_TIMETABLE_ID));
+
+      const formattedDownstreamResponse = {
+        ...resDown,
+        rollingStock: stdcmRollingStock,
+        creationDate: new Date(),
+        speedLimitByTag: osrdconf.speedLimitByTag,
+        simulationPathSteps: osrdconf.stdcmPathSteps,
+        path: resDown.status === 'conflicts' ? resDown.pathfinding_result : resDown.path,
+        alternativePath: true,
+      } as StdcmResponse;
+
+      const formattedUpstreamResponse = {
+        ...resUp,
+        rollingStock: stdcmRollingStock,
+        creationDate: new Date(),
+        speedLimitByTag: osrdconf.speedLimitByTag,
+        simulationPathSteps: osrdconf.stdcmPathSteps,
+        path: resUp.status === 'conflicts' ? resUp.pathfinding_result : resUp.path,
+        alternativePath: true,
+        upstreamPath: true,
+      } as StdcmResponse;
+
+      const formattedConflictResponse = {
+        ...response,
+        rollingStock: stdcmRollingStock,
+        creationDate: new Date(),
+        speedLimitByTag: osrdconf.speedLimitByTag,
+        simulationPathSteps: osrdconf.stdcmPathSteps,
+        path: response.pathfinding_result,
+        alternativePath: false,
+      } as StdcmConflictsResponse;
+
+      const upstreamInputs = {
+        ...currentSimulationInputs,
+        pathSteps: currentSimulationInputs.pathSteps.map((step) =>
+          adjustInputStepTolerances(step, UPSTREAM_TOLERANCE_SECONDS)
+        ),
+      };
+
+      const downtreamInputs = {
+        ...currentSimulationInputs,
+        pathSteps: currentSimulationInputs.pathSteps.map((step) =>
+          adjustInputStepTolerances(step, DOWNSTREAM_TOLERANCE_SECONDS)
+        ),
+      };
+
+      const currentSimulation = await createSimulation(
+        simulationsList.length + 1,
+        currentSimulationInputs,
+        formattedConflictResponse,
+        stdcmTrain,
+        { alternativePath: false, upstreamPath: true }
+      );
+      const updatedUpstreamTrain = updateTrainStartTime(stdcmTrain, formattedUpstreamResponse);
+      const updatedDownstreamTrain = updateTrainStartTime(stdcmTrain, formattedDownstreamResponse);
+
+      const downstreamSimulation = await createSimulation(
+        simulationsList.length + 3,
+        downtreamInputs,
+        formattedDownstreamResponse,
+        updatedDownstreamTrain,
+        { alternativePath: true, upstreamPath: false }
+      );
+
+      const upstreamSimulation = await createSimulation(
+        simulationsList.length + 2,
+        upstreamInputs,
+        formattedUpstreamResponse,
+        updatedUpstreamTrain,
+        { alternativePath: true, upstreamPath: true }
+      );
```

Comment:
You can use the function createSimulation here too
```suggestion
      const [currentSimulation, downstreamSimulation, upstreamSimulation] = ...
```